src/gui/plugins/transform_control/TransformControl.qml

```diff
@@ -339,153 +339,41 @@ ToolBar {
-        }
-        IgnSpinBox {
-          id: zEntry
-          minimumValue: 0.01
-          maximumValue: 100.0
-          decimals: 2
-          stepSize: 0.01
-          value: 1
-          Layout.row: 3
-          Layout.column: 1
-          onEditingFinished: {
-            TransformControl.OnSnapUpdate(
-              xEntry.value, yEntry.value, zEntry.value,
-              rollEntry.value, pitchEntry.value, yawEntry.value,
-              0, 0, 0
-            )
-          }
-        }
-        Text {
-          text: "Rotation (deg)"
+          text: "Rotation (rad)"
           font.weight: Font.Bold
           color: snapTitle
           Layout.columnSpan: 2
           Layout.row: 0
           Layout.column: 2
           bottomPadding: 10
         }
-        Text {
-          text: "Roll"
-          color: snapItem
-          Layout.row: 1
-          Layout.column: 2
-        }
-        IgnSpinBox {
-          id: rollEntry
-          minimumValue: 0.01
-          maximumValue: 180.0
-          decimals: 2
-          stepSize: 0.01
-          value: 45
+
+        GzPose {
+          id: gzPoseInstance
+          width: parent.width
+          Layout.columnSpan: 6
           Layout.row: 1
-          Layout.column: 3
-          onEditingFinished: {
-            TransformControl.OnSnapUpdate(
-              xEntry.value, yEntry.value, zEntry.value,
-              rollEntry.value, pitchEntry.value, yawEntry.value,
-              0, 0, 0
-            )
-          }
-        }
-        Text {
-          text: "Pitch"
-          color: snapItem
-          Layout.row: 2
-          Layout.column: 2
-        }
-        IgnSpinBox {
-          id: pitchEntry
-          minimumValue: 0.01
-          maximumValue: 180.0
-          decimals: 2
-          stepSize: 0.01
-          value: 45
-          Layout.row: 2
-          Layout.column: 3
-          onEditingFinished: {
-            TransformControl.OnSnapUpdate(
-              xEntry.value, yEntry.value, zEntry.value,
-              rollEntry.value, pitchEntry.value, yawEntry.value,
-              0, 0, 0
-            )
-          }
-        }
-        Text {
-          text: "Yaw"
-          color: snapItem
-          Layout.row: 3
-          Layout.column: 2
-        }
-        IgnSpinBox {
-          id: yawEntry
-          minimumValue: 0.01
-          maximumValue: 180.0
-          decimals: 2
-          stepSize: 0.01
-          value: 45
-          Layout.row: 3
-          Layout.column: 3
-          onEditingFinished: {
-            TransformControl.OnSnapUpdate(
-              xEntry.value, yEntry.value, zEntry.value,
-              rollEntry.value, pitchEntry.value, yawEntry.value,
-              0, 0, 0
-            )
+          Layout.fillWidth: true
+          readOnly: false
+          useRadian: false
+          spinMax: 180
+
+          xValue: 1.0
+          yValue: 1.0
+          zValue: 1.0
+          rollValue: 45
+          pitchValue: 45
+          yawValue: 45
```

Comment:
nit: let's move `useRadian` from above to here with the  `rpy` elements for better clarity
```suggestion
          useRadian: false
          rollValue: 45
          pitchValue: 45
          yawValue: 45
```